### PR TITLE
Changes for 1.16 and future

### PR DESCRIPTION
**Pull request by lucasreed**
- Updated a few daemonset templates to use apps/v1
- Updated some ingress objects to use networking.k8s.io/v1beta1 (not
breaking in 1.16, but will in 1.20)
- Updated chart minor versions for all modified

local testing results below:

With changes made, launched into v1.15.3 cluster:
 ✔︎ basic-demo => (version: "0.3.0", path: "incubator/basic-demo")
 ✔︎ fairwinds-metrics => (version: "0.4.0", path: "incubator/fairwinds-metrics")
 ✔︎ fluentd => (version: "4.1.0", path: "incubator/fluentd")
 ✔︎ logentries => (version: "0.2.0", path: "incubator/logentries")
 ✔︎ aws-iam-authenticator => (version: "v1.3.0", path: "stable/aws-iam-authenticator")
 ✔︎ polaris => (version: "0.10.0", path: "stable/polaris")

With changes made, launched into v1.16.4 cluster:
 ✔︎ basic-demo => (version: "0.3.0", path: "incubator/basic-demo")
 ✔︎ fairwinds-metrics => (version: "0.4.0", path: "incubator/fairwinds-metrics")
 ✔︎ fluentd => (version: "4.1.0", path: "incubator/fluentd")
 ✔︎ logentries => (version: "0.2.0", path: "incubator/logentries")
 ✔︎ aws-iam-authenticator => (version: "v1.3.0", path: "stable/aws-iam-authenticator")
 ✔︎ polaris => (version: "0.10.0", path: "stable/polaris")

Current repository (No changes other than Chart version) launched into v1.16.4 cluster:
 ✔︎ basic-demo => (version: "0.3.0", path: "incubator/basic-demo")
 ✔︎ fairwinds-metrics => (version: "0.4.0", path: "incubator/fairwinds-metrics")
 ✖︎ fluentd => (version: "4.1.0", path: "incubator/fluentd") > Error waiting for process: exit status 1
 ✖︎ logentries => (version: "0.2.0", path: "incubator/logentries") > Error waiting for process: exit status 1
 ✖︎ aws-iam-authenticator => (version: "v1.3.0", path: "stable/aws-iam-authenticator") > Error waiting for process: exit status 1
 ✔︎ polaris => (version: "0.10.0", path: "stable/polaris")

### CODE REVIEW

**Comment by lucasreed** [comment]
Important to note that the change to the ingress API is not a hard requirement for 1.16, and the new API version is only available in 1.14+. As I'm writing this I realize I should update the documentation for those charts. I'll do that.

File: stable/aws-iam-authenticator/templates/daemonset.yaml [hunk 1]
```diff
@@ -1,22 +1,30 @@
...
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ include "aws-iam-authenticator.chart" . }}
   annotations:
     checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
 spec:
   updateStrategy:
     type: RollingUpdate
+  selector:
+    matchLabels:
```
Comment (sudermanjr) [inline stable/aws-iam-authenticator/templates/daemonset.yaml:17]: Is this change necessary for the new API version? I'm not sure I've seen this before.

**Review by sudermanjr — APPROVED** [review]
This is awesome. Thank you for the thorough testing!

File: stable/aws-iam-authenticator/templates/daemonset.yaml (hunk 1 above)
Comment (lucasreed) [reply stable/aws-iam-authenticator/templates/daemonset.yaml:17]: It was failing e2e tests without this on my local machine. I saw where this is in the CONTRIBUTING doc as a standard so I changed it. Should I add back the old label `k8s-app`?

**Review by mjhuber — APPROVED** [review]
super awesome, thanks for updating these!!!

amazing

File: stable/aws-iam-authenticator/templates/daemonset.yaml (hunk 1 above)
Comment (sudermanjr) [reply stable/aws-iam-authenticator/templates/daemonset.yaml:17]: It's fine with me.

**Comment by rbren** [comment]
Given that `1.14` is so recent, does it make sense to gate `apiVersion` on `.Capabilities.KubeVersion.Major` and ` .Capabilities.KubeVersion.Minor`?

e.g. 
```
{{ if and (eq .Capabilities.KubeVersion.Major 1) (lt .Capabilities.KubeVersion.Minor 14) }}
apiVersion: extensions/v1beta1
{{ else }}
apiVersion: networking.k8s.io/v1beta1
{{ end }}
```

**Comment by rbren** [comment]
(no strong opinions there - just curious)

**Review by rbren — APPROVED** [review]

**Comment by lucasreed** [comment]
> Given that `1.14` is so recent, does it make sense to gate `apiVersion` on `.Capabilities.KubeVersion.Major` and ` .Capabilities.KubeVersion.Minor`?
> 
> e.g.
> 
> ```
> {{ if and (eq .Capabilities.KubeVersion.Major 1) (lt .Capabilities.KubeVersion.Minor 14) }}
> apiVersion: extensions/v1beta1
> {{ else }}
> apiVersion: networking.k8s.io/v1beta1
> {{ end }}
> ```

Andy and I spoke about this, and our initial thought was to leave it the way we have it now because 1.14 is out of support band.

**Comment by sudermanjr** [comment]
Yep, according to their policy, the last three versions are supported. That means 1.15-1.17 right now.  1.14 is now almost a year old, which for Kube is ancient history. :-D 

https://kubernetes.io/docs/setup/release/version-skew-policy/#supported-versions

**Review by makoscafee — APPROVED** [review]
